sdk-python/littlehorse/workflow.py

```diff
@@ -459,32 +612,74 @@ def add_node(self, name: str, sub_node: NodeType) -> str:
             raise TypeError("The first node should be a EntrypointNode")
 
         if len(self._nodes) > 0:
-            previous_node = self._nodes[list(self._nodes)[-1]]
-            previous_node.outgoing_edges.append(Edge(sink_node_name=next_node_name))
-
-            # TODO add node condition
-
-        if node_type == NodeCase.TASK:
-            self._nodes[next_node_name] = Node(task=sub_node)  # type: ignore[arg-type]  # noqa: E501
-        if node_type == NodeCase.ENTRYPOINT:
-            self._nodes[next_node_name] = Node(entrypoint=sub_node)  # type: ignore[arg-type]  # noqa: E501
-        if node_type == NodeCase.EXIT:
-            self._nodes[next_node_name] = Node(exit=sub_node)  # type: ignore[arg-type]  # noqa: E501
-        if node_type == NodeCase.EXTERNAL_EVENT:
-            self._nodes[next_node_name] = Node(external_event=sub_node)  # type: ignore[arg-type]  # noqa: E501
-        if node_type == NodeCase.SLEEP:
-            self._nodes[next_node_name] = Node(sleep=sub_node)  # type: ignore[arg-type]  # noqa: E501
-        if node_type == NodeCase.START_THREAD:
-            self._nodes[next_node_name] = Node(start_thread=sub_node)  # type: ignore[arg-type]  # noqa: E501
-        if node_type == NodeCase.WAIT_FOR_THREADS:
-            self._nodes[next_node_name] = Node(wait_for_threads=sub_node)  # type: ignore[arg-type]  # noqa: E501
-        if node_type == NodeCase.NOP:
-            self._nodes[next_node_name] = Node(nop=sub_node)  # type: ignore[arg-type]  # noqa: E501
-        if node_type == NodeCase.USER_TASK:
-            self._nodes[next_node_name] = Node(user_task=sub_node)  # type: ignore[arg-type]  # noqa: E501
+            last_node = self._last_node()
+            last_node.outgoing_edges.append(Edge(sink_node_name=next_node_name))
+
+        self._nodes.append(WorkflowNode(next_node_name, node_type, sub_node))
 
         return next_node_name
 
+    def condition(
+        self, left_hand: Any, comparator: Comparator, right_hand: Any
+    ) -> WorkflowCondition:
+        """Returns a WorkflowCondition that can be used in
+        `ThreadBuilder.doIf()` or `ThreadBuilder.doElse()`.
+
+        Args:
+            left_hand (Any): is either a literal value
+            (which the Library casts to a Variable Value) or a
+            `WfRunVariable` representing the LHS of the expression.
+            comparator (Comparator): is a Comparator defining the
+            comparator, for example, `ComparatorTypePb.EQUALS`.
+            right_hand (Any): is either a literal value
+            (which the Library casts to a Variable Value) or a
+            `WfRunVariable` representing the RHS of the expression.
+
+        Returns:
+            WorkflowCondition: a WorkflowCondition.
+        """
+        return WorkflowCondition(left_hand, comparator, right_hand)
+
+    def do_if(
+        self, condition: WorkflowCondition, initializer: "ThreadInitializer"
```

Comment:
`ThreadInitializer` is much better than `ThreadBuilder`. Perhaps it's worth changing in the other SDK's?

In Go, changing the names won't be hard at all since it uses duck-typing.

In Java, we could have `public interface ThreadInitializer`, and `public interface ThreadBuilder extends ThreadInitializer` but `ThreadBuilder` doesn't have anything different. That way maybe the current API contract works, and we can mark `ThreadBuilder` as `@Deprecated`